projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/return-order-confirmation/return-order-confirmation.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, OnInit } from '@angular/core';
+import { formatCurrency, getCurrencySymbol } from '@angular/common';
+import { Observable, combineLatest } from 'rxjs';
+import { tap, filter, map } from 'rxjs/operators';
+import {
+  Order,
+  OrderEntry,
+  CancellationReturnRequestEntryInput,
+  RoutingService,
+  LanguageService,
+} from '@spartacus/core';
+
+import { OrderDetailsService } from '../../order-details/order-details.service';
+
+@Component({
+  selector: 'cx-return-order-confirmation',
+  templateUrl: './return-order-confirmation.component.html',
+})
+export class ReturnOrderConfirmationComponent implements OnInit {
+  constructor(
+    protected orderDetailsService: OrderDetailsService,
+    protected routing: RoutingService,
+    protected languageService: LanguageService
+  ) {}
+
+  order$: Observable<Order>;
+  returnedEntries: OrderEntry[] = [];
+  orderCode: string;
+
+  lang = 'en';
+  returnRequestEntryInput: CancellationReturnRequestEntryInput[];
+
+  ngOnInit() {
+    this.returnRequestEntryInput = this.orderDetailsService.cancellationReturnRequestInputs;
+
+    this.order$ = combineLatest([
+      this.orderDetailsService.getOrderDetails(),
+      this.languageService.getActive(),
+    ]).pipe(
+      filter(([order]) => Boolean(order.entries)),
+      tap(([order, lang]) => {
+        this.lang = lang;
+        this.orderCode = order.code;
+
+        this.returnedEntries = [];
+        order.entries.forEach(entry => {
+          const returnedQty = this.getEntryReturnedQty(entry);
+          if (returnedQty > 0) {
+            const copiedEntry = Object.assign({}, entry, {
+              returnedItemsPrice: null,
+              returnedQuantity: 0,
+            });
+            this.setReturnedEntryPrice(returnedQty, copiedEntry);
+            this.returnedEntries.push(copiedEntry);
```

Comment:
Instead of pushing this to `returnEntries` can't this first type be a map and return a new array of entries?

That way you can subscribe to order$ and use that value where you use returnedEntries and returnedEntries can be removed.